rust/src/participant.rs

```diff
@@ -102,42 +100,55 @@ impl Participant {
     /// Compose a sum message.
     pub fn compose_sum_message(&mut self, pk: &CoordinatorPublicKey) -> Vec<u8> {
         self.gen_ephm_keypair();
-        SumMessage::from_parts(
-            &self.pk,
-            &self.sum_signature,
-            &self.ephm_pk,
-            &self.certificate,
-        )
-        .seal(&self.sk, pk)
+
+        let payload = SumOwned {
+            sum_signature: self.sum_signature,
+            ephm_pk: self.ephm_pk,
+        };
+
+        let message = MessageOwned::new_sum(*pk, self.pk, payload);
+        self.seal_message(pk, &message)
     }
 
     /// Compose an update message.
-    pub fn compose_update_message(&self, pk: &CoordinatorPublicKey, sum_dict: &SumDict) -> Vec<u8> {
+    pub fn compose_update_message(&self, pk: CoordinatorPublicKey, sum_dict: &SumDict) -> Vec<u8> {
```

Comment:
signatures of the three `compose_` methods are not consistent anymore as they take `pk` by value or reference. i would prefer by reference, what's the reason to change it to by value?

Reply:
So I don't mind rolling this back. Note that api is inconsistent because I forgot to update `_compose_sum_message`. My goal was to have the `_compose` method take `pk` by value.

The reason is that internally `MessageOwned` takes the key by value. I could refrain from making that change because the public key is `Copy`, but I thought it was more explicit to have the caller perform the copy.

Reply:
my preference for by-reference comes from the usage in `sodiumoxide`, but calling it by-value would also be ok as long as it is consistent.

Reply:
I see. I'll undo these changes. It will also make the diff smaller, so less likely to conflict with your other PR.